### ISSUE TITLE
Server kicked me out with message: Outdated server!

### ISSUE THREAD

**Issue by kurtismullins** [lukleh/TwistedBot#5]
I updated to the latest Craftbukkit Development build (1.4.7-R1.1) to try to circumvent this error. Unfortunately, it didn't help.

I have no trouble with connecting from multiple clients of new (and relatively new) versions.

Is this an issue with the version being sent/accepted by TwistedBot or is this message legitimate? 

When I type:

```
python ./bot.py --serverhost 192.168.0.11
```

Here's the output:

```
03:38:52.402203 [-] Start logging
03:38:52.457013 [BLOCKS] registered 159 blocks
03:38:52.464399 [ITEMS] registered 263 items
03:38:52.529823 [RECIPES] registered 27 recipes
03:38:52.529934 [RECIPES] know how to brew 0 items
03:38:52.529979 [RECIPES] know how to mob kill 0 items
03:38:52.530018 [RECIPES] know how to craft 17 items
03:38:52.530056 [RECIPES] know how to smelt 3 items
03:38:52.530131 [RECIPES] know how to mine 9 items
03:38:52.546298 [MAIN] terminal chat available
03:38:52.546564 [PROTOCOL] Started connecting...
03:38:52.547282 [PROTOCOL] Connected!
03:38:52.548014 [PROTOCOL] sending Handshake
03:38:52.577646 [PROTOCOL] Server kicked me out with message: Outdated server!
03:38:52.577937 [WORLD] Shutdown
```

Thanks.

**Comment by lukleh** [comment]
Recent code runs against 13w07a snapshot. Minecraft server checks client protocol version, and if it does not match, it presumes the client is outdated, although it can be the other way around.
To answer your question: yes, the message is legitimate.
Do you need to run Craftbukkit 1.4.7 now? Try vanilla server (snapshot version).
When everything moves to Minecraft 1.5, there should be no difference.

**Comment by kurtismullins** [comment]
That makes sense. Thanks for clarifying that. I'll run a vanilla server for testing purposes.

I use plugins; hence the reason for craftbukkit. Is there any way to simulate an older minecraft version or has the protocol broken backwards compatibility?

**Comment by lukleh** [comment]
You are welcome.

The protocol is not backward compatible. Sometimes I go a little bit ahead of time and start to work with snapshots. Probably would be a better idea to keep it cool and stay with the stable mainline.

Anyway, in less than a month an official Minecraft 1.5 and Craftbukkit will be released (my estimate) and then the bot can be used more freely.

Can I ask you how are you using the bot?

**Comment by kurtismullins** [comment]
That's great news.

I actually want to experiment with scripting bots to do things such as auto mining and playing the role of a travelling merchant. I'm also interested in pathfinding and other AI algorithms. My young child plays on our private network server so I might do some cool things for kids as well.

I started out trying to implement my own bot but quickly realized that its a relatively high barrier to with all of the overhead. Luckily I stumbled across yours since you've already done a lot of the "dirty" work and you've written it in Python (my language of choice). Thanks for sharing it.
